### ISSUE TITLE
anbox PKGBUILD does not build

### ISSUE THREAD

**Issue by 3nprob** [dreemurrs-embedded/Pine64-Arch#327]
### Steps to reproduce
```
guzuta build --arch aarch64 --chroot-dir ./chroot-aarch64 --repo-name danctnix --repo-dir ./repo/danctnix/aarch64 --srcdest ./sources --logdest ./logs PKGBUILDS/danctnix/anbox
```

### Expected behavior
Package is built

### Actual behavior
Build  fails with compilation error

### Logfiles and additional information
```
:: Synchronizing package databases...
100%:: Starting full system upgrade...
 there is nothing to do
==> Making package: anbox 0_git20210625-2 (Tue Feb 22 18:17:03 2022)
==> Retrieving sources...
  -> Found anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8.tar.gz
  -> Found cpu_features-b5c271c53759b2b15ff91df19bd0b32f2966e275.tar.gz
  -> Found sdbus_cpp-3b735bf1aad65277f56e65c828a22455cbaf5245.tar.gz
  -> Found inputmethod-hack.patch
  -> Found give-more-time-to-start.patch
  -> Found 0001-Don-t-display-Android-apps-in-the-app-drawer.patch
  -> Found 1827.patch
  -> Found anbox-container-manager.service
  -> Found anbox-session-manager.service
  -> Found 99-anbox.rules
  -> Found anbox.desktop
  -> Found dev-binderfs.mount
==> Validating source files with md5sums...
    anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8.tar.gz ... Passed
    cpu_features-b5c271c53759b2b15ff91df19bd0b32f2966e275.tar.gz ... Passed
    sdbus_cpp-3b735bf1aad65277f56e65c828a22455cbaf5245.tar.gz ... Passed
    inputmethod-hack.patch ... Passed
    give-more-time-to-start.patch ... Passed
    0001-Don-t-display-Android-apps-in-the-app-drawer.patch ... Passed
    1827.patch ... Passed
    anbox-container-manager.service ... Passed
    anbox-session-manager.service ... Passed
    99-anbox.rules ... Passed
    anbox.desktop ... Passed
    dev-binderfs.mount ... Passed
==> Making package: anbox 0_git20210625-2 (Tue 22 Feb 2022 06:17:20 PM UTC)
==> Checking runtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
looking for conflicting packages...
warning: dependency cycle detected:
warning: mesa will be installed before its libglvnd dependency

Packages (38) giflib-5.2.1-2  hidapi-0.11.2-1  libdrm-2.4.110-1  libedit-20210910_3.1-1  libglvnd-1.4.0-1  libibus-1.5.25-5  libjpeg-turbo-2.1.2-1  libomxil-bellagio-0.9.3-3
              libpciaccess-0.16-2  libpng-1.6.37-3.1  libtiff-4.3.0-1  libunwind-1.6.2-1  libusb-1.0.25-2  libwebp-1.2.2-1.1  libx11-1.7.3.1-1  libxau-1.0.9-3  libxcb-1.14-1
              libxcursor-1.2.0-2  libxdamage-1.1.5-3  libxdmcp-1.1.3-3  libxext-1.3.4-3  libxfixes-6.0.0-1  libxrender-0.9.10-4  libxshmfence-1.3-2  libxxf86vm-1.1.4-4
              llvm-libs-13.0.1-1  lm_sensors-1:3.6.0.r41.g31d1f125-1  mesa-21.3.6-1  sdl2-2.0.20-2  vulkan-icd-loader-1.2.203-1  wayland-1.20.0-1  wget-1.21.2-1  xcb-proto-1.14.1-5
              xorgproto-2021.5-1  boost-libs-1.78.0-1  lxc-1:4.0.12-1  protobuf-3.19.4-1  sdl2_image-2.0.5-2

Total Installed Size:  346.18 MiB

:: Proceed with installation? [Y/n]
(38/38) checking keys in keyring                                                                             [#################################################################] 100%
(38/38) checking package integrity                                                                           [#################################################################] 100%
(38/38) loading package files                                                                                [#################################################################] 100%
(38/38) checking for file conflicts                                                                          [#################################################################] 100%
(38/38) checking available disk space                                                                        [#################################################################] 100%
:: Processing package changes...
( 1/38) installing wget                                                                                      [#################################################################] 100%
Optional dependencies for wget
    ca-certificates: HTTPS downloads [installed]
( 2/38) installing lxc                                                                                       [#################################################################] 100%
Optional dependencies for lxc
    dnsmasq: lxc-net.service
    lua
    lua-filesystem: lxc-top    lua-alt-getopt: lxc-top( 3/38) installing xcb-proto                                                                                 [#################################################################] 100%
( 4/38) installing libxdmcp                                                                                  [#################################################################] 100%
( 5/38) installing libxau                                                                                    [#################################################################] 100%
( 6/38) installing libxcb                                                                                    [#################################################################] 100%
( 7/38) installing xorgproto                                                                                 [#################################################################] 100%
( 8/38) installing libx11                                                                                    [#################################################################] 100%
( 9/38) installing libxext                                                                                   [#################################################################] 100%
(10/38) installing libxrender                                                                                [#################################################################] 100%
(11/38) installing libpciaccess                                                                              [#################################################################] 100%
(12/38) installing libdrm                                                                                    [#################################################################] 100%
(13/38) installing wayland                                                                                   [#################################################################] 100%
(14/38) installing libxxf86vm                                                                                [#################################################################] 100%
(15/38) installing libxfixes                                                                                 [#################################################################] 100%
(16/38) installing libxdamage                                                                                [#################################################################] 100%
(17/38) installing libxshmfence                                                                              [#################################################################] 100%
(18/38) installing libomxil-bellagio                                                                         [#################################################################] 100%
(19/38) installing libunwind                                                                                 [#################################################################] 100%
(20/38) installing libedit                                                                                   [#################################################################] 100%
(21/38) installing llvm-libs                                                                                 [#################################################################] 100%
(22/38) installing lm_sensors                                                                                [#################################################################] 100%
Optional dependencies for lm_sensors
    rrdtool: for logging with sensord
    perl: for sensor detection and configuration convert [installed]
(23/38) installing vulkan-icd-loader                                                                         [#################################################################] 100%
Optional dependencies for vulkan-icd-loader
    vulkan-driver: packaged vulkan driver
(24/38) installing mesa                                                                                      [#################################################################] 100%
Optional dependencies for mesa
    opengl-man-pages: for the OpenGL API man pages
    mesa-vdpau: for accelerated video playback
    libva-mesa-driver: for accelerated video playback
(25/38) installing libglvnd                                                                                  [#################################################################] 100%
(26/38) installing libxcursor                                                                                [#################################################################] 100%
Optional dependencies for libxcursor
    gnome-themes-standard: fallback icon theme
(27/38) installing libibus                                                                                   [#################################################################] 100%
(28/38) installing hidapi                                                                                    [#################################################################] 100%
Optional dependencies for hidapi
    libusb: for the libusb backend -- hidapi-libusb.so [pending]
    libudev.so: for the hidraw backend -- hidapi-hidraw.so [installed]
(29/38) installing libusb                                                                                    [#################################################################] 100%
(30/38) installing sdl2                                                                                      [#################################################################] 100%
Optional dependencies for sdl2
    alsa-lib: ALSA audio driver
    libpulse: PulseAudio audio driver
    jack: JACK audio driver
    pipewire: PipeWire audio driver
    libdecor: Wayland client decorations
(31/38) installing libpng                                                                                    [#################################################################] 100%
(32/38) installing libjpeg-turbo                                                                             [#################################################################] 100%
(33/38) installing libtiff                                                                                   [#################################################################] 100%
Optional dependencies for libtiff
    freeglut: for using tiffgt
(34/38) installing giflib                                                                                    [#################################################################] 100%
(35/38) installing libwebp                                                                                   [#################################################################] 100%
Optional dependencies for libwebp
    freeglut: vwebp viewer
(36/38) installing sdl2_image                                                                                [#################################################################] 100%
(37/38) installing protobuf                                                                                  [#################################################################] 100%
(38/38) installing boost-libs                                                                                [#################################################################] 100%Optional dependencies for boost-libs    openmpi: for mpi support
:: Running post-transaction hooks...
(1/4) Reloading system manager configuration...
  Skipped: Current root is not booted.
(2/4) Creating temporary files...
(3/4) Arming ConditionNeedsUpdate...
(4/4) Updating the info directory file...
==> Checking buildtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
looking for conflicting packages...

Packages (13) hicolor-icon-theme-0.17-2  jsoncpp-1.9.5-1  libuv-1.43.0-1  perl-error-0.17029-3  perl-mailtools-2.21-5  perl-timedate-2.33-3  rhash-1.4.2-1  boost-1.78.0-1
              cmake-3.22.2-1  git-2.35.1-1  glm-0.9.9.8-1  gtest-1.11.0-3  properties-cpp-0.0.1-1
Total Installed Size:  304.33 MiB

:: Proceed with installation? [Y/n]
(13/13) checking keys in keyring                                                                             [#################################################################] 100%
(13/13) checking package integrity                                                                           [#################################################################] 100%
(13/13) loading package files                                                                                [#################################################################] 100%
(13/13) checking for file conflicts                                                                          [#################################################################] 100%
(13/13) checking available disk space                                                                        [#################################################################] 100%
:: Processing package changes...
( 1/13) installing hicolor-icon-theme                                                                        [#################################################################] 100%
( 2/13) installing jsoncpp                                                                                   [#################################################################] 100%
Optional dependencies for jsoncpp
    jsoncpp-doc: documentation
( 3/13) installing libuv                                                                                     [#################################################################] 100%
( 4/13) installing rhash                                                                                     [#################################################################] 100%
( 5/13) installing cmake                                                                                     [#################################################################] 100%
Optional dependencies for cmake
    qt6-base: cmake-gui
( 6/13) installing perl-error                                                                                [#################################################################] 100%
( 7/13) installing perl-timedate                                                                             [#################################################################] 100%
( 8/13) installing perl-mailtools                                                                            [#################################################################] 100%
( 9/13) installing git                                                                                       [#################################################################] 100%
Optional dependencies for git
    tk: gitk and git gui
    perl-libwww: git svn
    perl-term-readkey: git svn and interactive.singlekey setting
    perl-io-socket-ssl: git send-email TLS support
    perl-authen-sasl: git send-email TLS support
    perl-mediawiki-api: git mediawiki support
    perl-datetime-format-iso8601: git mediawiki support
    perl-lwp-protocol-https: git mediawiki https support
    perl-cgi: gitweb (web interface) support
    python: git svn & git p4 [installed]
    subversion: git svn
    org.freedesktop.secrets: keyring credential helper
    libsecret: libsecret credential helper [installed]
(10/13) installing glm                                                                                       [#################################################################] 100%
(11/13) installing properties-cpp                                                                            [#################################################################] 100%
(12/13) installing boost                                                                                     [#################################################################] 100%
Optional dependencies for boost
    python: for python bindings [installed]
(13/13) installing gtest                                                                                     [#################################################################] 100%
Optional dependencies for gtest
    python: gmock generator [installed]
:: Running post-transaction hooks...(1/4) Creating system user accounts...Creating group 'git' with GID 975.
Creating user 'git' (git daemon user) with UID 975 and GID 975.
(2/4) Reloading system manager configuration...
  Skipped: Current root is not booted.
(3/4) Arming ConditionNeedsUpdate...
(4/4) Warn about old perl modules
==> Retrieving sources...
  -> Found anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8.tar.gz
  -> Found cpu_features-b5c271c53759b2b15ff91df19bd0b32f2966e275.tar.gz
  -> Found sdbus_cpp-3b735bf1aad65277f56e65c828a22455cbaf5245.tar.gz
  -> Found inputmethod-hack.patch
  -> Found give-more-time-to-start.patch
  -> Found 0001-Don-t-display-Android-apps-in-the-app-drawer.patch
  -> Found 1827.patch
  -> Found anbox-container-manager.service
  -> Found anbox-session-manager.service
  -> Found 99-anbox.rules
  -> Found anbox.desktop
  -> Found dev-binderfs.mount
==> WARNING: Skipping all source file integrity checks.
==> Extracting sources...
  -> Extracting anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8.tar.gz with bsdtar
  -> Extracting cpu_features-b5c271c53759b2b15ff91df19bd0b32f2966e275.tar.gz with bsdtar
  -> Extracting sdbus_cpp-3b735bf1aad65277f56e65c828a22455cbaf5245.tar.gz with bsdtar
==> Starting prepare()...
patching file src/anbox/graphics/multi_window_composer_strategy.cpp
Hunk #1 succeeded at 25 (offset -1 lines).
patching file src/anbox/graphics/multi_window_composer_strategy.h
Hunk #2 succeeded at 33 with fuzz 2 (offset -1 lines).
patching file src/anbox/cmds/launch.cpp
Hunk #1 succeeded at 35 (offset 1 line).
patching file src/anbox/application/launcher_storage.cpp
patching file src/anbox/container/lxc_container.cpp
==> Starting build()...
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found Boost: /usr/lib/cmake/Boost-1.78.0/BoostConfig.cmake (found version "1.78.0") found components: filesystem log serialization system thread program_options
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0")
-- Found Threads: TRUE
-- Found EGL: /usr/lib/libEGL.so
-- Found GLESv2: /usr/lib/libGLESv2.so
-- Found Protobuf: /usr/lib/libprotobuf.so (found version "3.19.4")-- Checking for module 'sdl2'--   Found sdl2, version 2.0.20
-- Checking for module 'SDL2_image'
--   Found SDL2_image, version 2.0.5
-- Checking for module 'dbus-1'
--   Found dbus-1, version 1.12.20
-- Checking for module 'lxc'
--   Found lxc, version 4.0.12
-- Checking for module 'properties-cpp'
--   Found properties-cpp, version 0.0.1
-- Checking for module 'libsystemd'
--   Found libsystemd, version 250
-- Checking for module 'libcap'
--   Found libcap, version 2.63
-- LXC version: 4.0.12
CMake Deprecation Warning at external/process-cpp-minimal/CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Found Boost: /usr/lib/cmake/Boost-1.78.0/BoostConfig.cmake (found version "1.78.0") found components: iostreams system
-- Found Boost: /usr/lib/cmake/Boost-1.78.0/BoostConfig.cmake (found version "1.78.0") found components: filesystem system unit_test_framework
-- Found libdw: /usr/lib/libdw.so
-- Found libbfd: /usr/lib/libbfd.so
-- Could NOT find libdwarf (missing: LIBDWARF_LIBRARY LIBDWARF_INCLUDE_DIR)
-- Found Backward: /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/backward-cpp
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for getauxval
-- Looking for getauxval - found
-- Configuring done
-- Generating done
-- Build files have been written to: /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/build
[  0%] Creating directories for 'sdbus-cpp'
[  0%] No download step for 'sdbus-cpp'
[  1%] No update step for 'sdbus-cpp'
[  1%] No patch step for 'sdbus-cpp'
[  2%] Performing configure step for 'sdbus-cpp'
loading initial cache file /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/build/external/sdbus-cpp/tmp/sdbus-cpp-cache-None.cmake
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0")
-- Checking for module 'libsystemd>=236'
--   Found libsystemd, version 250
-- Building with code generator tool
-- Found EXPAT: /usr/lib/libexpat.so (found version "2.4.5")
-- Building with documentation
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)-- Configuring done-- Generating done
-- Build files have been written to: /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/build/external/sdbus-cpp/src/sdbus-cpp-build
[  2%] Performing build step for 'sdbus-cpp' 
[...]
[ 31%] Generating GLESv1_dec/renderControl_dec.cpp, GLESv1_dec/renderControl_server_context.cpp
[ 32%] Building CXX object external/android-emugl/host/libs/CMakeFiles/renderControl_dec.dir/GLESv1_dec/renderControl_dec.cpp.o
[ 32%] Building CXX object external/android-emugl/host/libs/CMakeFiles/renderControl_dec.dir/GLESv1_dec/renderControl_server_context.cpp.o
[ 32%] Linking CXX static library librenderControl_dec.a
[ 32%] Built target renderControl_dec
[ 32%] Building CXX object external/xdg/CMakeFiles/xdg.dir/xdg.cpp.o
In file included from /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.cpp:16:
/build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.h:94:17: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
   94 |     static std::shared_ptr<BaseDirSpecification> create();
      |                 ^~~~~~~~~~
/build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.h:19:1: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory
>’?
   18 | #include <boost/filesystem.hpp>
  +++ |+#include <memory>
   19 |
/build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.cpp:179:44: error: no declaration matches ‘std::shared_ptr<xdg::BaseDirSpecification> xdg::BaseDirSp
ecification::create()’
  179 | std::shared_ptr<xdg::BaseDirSpecification> xdg::BaseDirSpecification::create()
      |                                            ^~~
/build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.cpp:179:44: note: no functions named ‘std::shared_ptr<xdg::BaseDirSpecification> xdg::BaseDirSpecifi
cation::create()’
In file included from /build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.cpp:16:
/build/anbox/src/anbox-04ac697c890b96aa70814a29696adee6a2fc7ff8/external/xdg/xdg.h:90:7: note: ‘class xdg::BaseDirSpecification’ defined here
   90 | class BaseDirSpecification : NotCopyable, NotMoveable
      |       ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [external/xdg/CMakeFiles/xdg.dir/build.make:76: external/xdg/CMakeFiles/xdg.dir/xdg.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:672: external/xdg/CMakeFiles/xdg.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Build failed, check /home/user/dev/Pine64-Arch/chroot-aarch64/user/build
Error: Unable to build package in PKGBUILDS/danctnix/anbox

Caused by:
    makechrootpkg failed
```

**Comment by 3nprob** [comment]
Looks like this issue appears when building with glibc >= 2.34 and fixed in https://github.com/anbox/anbox/commit/4ef88ab7005733480e88a4aab71de23f27c49842

**Comment by Danct12** [comment]
Fixed in c0e8a40e30260fc2fd9085a0edead270b776e14c
